sled-agent/src/rack_setup/service.rs

```diff
@@ -733,26 +824,62 @@ impl ServiceInner {
         );
 
         // Convert all the information we have about datasets into a format
-        // which can be processed by Nexus.
-        let mut datasets: Vec<NexusTypes::DatasetCreateRequest> = vec![];
+        // which can be stored in CockroachDB once they make their way to Nexus.
+        //
+        // TODO(https://github.com/oxidecomputer/omicron/issues/6998): Remove me!
+        // This is somewhat redundant with the information we're supplying in
+        // the initial blueprint, and could be de-duplicated if we can fully
+        // migrate to a world where "datasets exist in the blueprint, but not
+        // in CockroachDB".
+        let mut datasets = HashMap::<
+            (ZpoolUuid, DatasetKind),
+            NexusTypes::DatasetCreateRequest,
+        >::new();
         for sled_config in service_plan.services.values() {
-            for zone in &sled_config.zones {
-                if let Some(dataset) = zone.zone_type.durable_dataset() {
-                    datasets.push(NexusTypes::DatasetCreateRequest {
-                        zpool_id: dataset
-                            .dataset
-                            .pool_name
-                            .id()
-                            .into_untyped_uuid(),
-                        dataset_id: zone.id.into_untyped_uuid(),
```

Comment:
This was nasty - we re-used the zone UUID as a dataset UUID :grimacing: 

We aren't doing that anymore - datasets have their own Uuids that are provisioned as they are created within the plan.